### ISSUE TITLE
Support Multiple HTTPS Sites in one VM 

### ISSUE THREAD

**Issue by aweingarten** [geerlingguy/drupal-vm#321]
Followed the instructions here 
http://docs.drupalvm.com/en/latest/deployment/multisite/
and 
https://github.com/geerlingguy/ansible-role-apache

To add multiple HTTPS sites to a single VM.  

When I did so I found that when I tried to access the second apache_vhosts_ssl entry it loaded the source from the first entry. 

Restarting Apache inside the vm resulted in the following:

```
* Restarting web server apache2                                                                                                                                         [Thu Nov 19 22:08:03 2015] [warn] ​_default_​ VirtualHost overlap on port 443, the first has precedence
... waiting [Thu Nov 19 22:08:04 2015] [warn] ​_default_​ VirtualHost overlap on port 443, the first has precedence
```

A work around was to open 
`/etc/apache2/ports.conf`

Add the line:

```
NameVirtualHost *:443
```

right above the Listen 443 line
And then restart Apache

**Comment by geerlingguy** [comment]
Thanks for reporting this! I think I'll have to look into the defaults across CentOS 7 and Ubuntu 14.04 and see what needs to be done in the `geerlingguy.apache` role to make sure the vhosts are all available by default.

**Comment by aweingarten** [comment]
@geerlingguy, one other wrinkle to report this was on ubuntu 12.  I can retest with 14.  I'm assuming that the issue will be the same.

**Comment by greylabel** [comment]
@aweingarten adding an `apache_vhosts_ssl` entry for each worked for me:

```
apache_vhosts_ssl:
  - servername: "local.site1.example.com"
    documentroot: "{{ drupal_core_path }}"
    certificate_file: "/var/www/drupalvm/box/examples/ssl/local_star_example_com.crt"
    certificate_key_file: "/var/www/drupalvm/box/examples/ssl/local_star_example_com.key"
    extra_parameters: |
          ProxyPassMatch ^/(.*\.php(/.*)?)$ "fcgi://127.0.0.1:9000{{ drupal_core_path }}"

apache_vhosts_ssl:
  - servername: "local.site2.example.com"
    documentroot: "{{ drupal_core_path }}"
    certificate_file: "/var/www/drupalvm/box/examples/ssl/local_star_example_com.crt"
    certificate_key_file: "/var/www/drupalvm/box/examples/ssl/local_star_example_com.key"
    extra_parameters: |
          ProxyPassMatch ^/(.*\.php(/.*)?)$ "fcgi://127.0.0.1:9000{{ drupal_core_path }}"
```

**Comment by geerlingguy** [comment]
@greylabel - Are you sure that worked? If you define the same variable twice, the second definition should override the first, and you would (in the example above) only be able to access `local.site2.example.com`.

**Comment by oxyc** [comment]
On Ubuntu 16.04 it works to define multiple vhosts like so:

```yaml
apache_vhosts_ssl:
  - servername: "local.site1.example.com"
    documentroot: "{{ drupal_core_path }}"
    certificate_file: "/var/www/drupalvm/box/examples/ssl/local_star_example_com.crt"
    certificate_key_file: "/var/www/drupalvm/box/examples/ssl/local_star_example_com.key"
    extra_parameters: |
          ProxyPassMatch ^/(.*\.php(/.*)?)$ "fcgi://127.0.0.1:9000{{ drupal_core_path }}"
  - servername: "local.site2.example.com"
    documentroot: "{{ drupal_core_path }}"
    certificate_file: "/var/www/drupalvm/box/examples/ssl/local_star_example_com.crt"
    certificate_key_file: "/var/www/drupalvm/box/examples/ssl/local_star_example_com.key"
    extra_parameters: |
          ProxyPassMatch ^/(.*\.php(/.*)?)$ "fcgi://127.0.0.1:9000{{ drupal_core_path }}"
```

This wasn't dry enough so I added some extreme magic that automatically add the snakeoil certificate to all previously defined apache_vhosts

```yaml
apache_vhost_ssl_addition:
  certificate_file: "/etc/ssl/certs/ssl-cert-snakeoil.pem"
  certificate_key_file: "/etc/ssl/private/ssl-cert-snakeoil.key"

# Generate a SSL virtual host for every regular vhost.
apache_vhosts_ssl: "{% set vhosts = [] %}{% for vhost in apache_vhosts %}{% if vhosts.append(vhost|combine(apache_vhost_ssl_addition)) %}{% endif %}{% endfor %}{{ vhosts }}"
```

**Comment by oxyc** [comment]
@aweingarten did you ever re-test on Ubuntu 14.04? As this issue was reported before https://github.com/geerlingguy/drupal-vm/issues/589, I'm thinking it might be fixed now.

**Comment by oxyc** [comment]
I'm fairly certain this was fixed by #589. I just tested it on Ubuntu 12.04 and it works fine. @geerlingguy we don't support Apache 2.2 do we?

**Comment by geerlingguy** [comment]
I think I still kinda-sorta support it, but it's annoying to use 2.2 (for Apache role, you'd have to set `apache_vhosts_version` and make sure other bits of config are correct still.

I'm happy to drop 2.2 support entirely at this point, though.

**Comment by oxyc** [comment]
Well with the defaut configs this issue is fixed at least. I'm guessing there are more features that do not work with 2.2 anyways. Do you think we can close this as fixed?

**Comment by geerlingguy** [comment]
Yes, let's. I'll let you do the honors!

**Comment by oxyc** [comment]
Yey, fixed by #589.
